### PR TITLE
add `cranelift-codegen-0.82.1` benchmark

### PR DESCRIPTION
**Pull request by lqd**
From-scratch build times are faster so this should be good for CI/bot:

|   | cranelift-codegen  |  cranelift-codegen-0.82.1 |
|---|---:|---:|
| check  | 13.92s  | 8.61s  |
| debug  | 17.91s  |  11.96s |
| opt  | 43.09s  |  16.38s |

The leaf crate itself has changed a lot (like bjorn3 mentioned elsewhere) but is still close enough, and more importantly hasn't turned into a helloworld façade. Eyeballing the recent version to be around 1.5x slower over the 3 profiles. 
image

It seems fine to remove the old one in the future ?

### CODE REVIEW

**Comment by nnethercote** [comment]
Thank you for doing this, I will review it in my morning.

It's good that the overall time is smaller but the final crate time is bigger. And removing the old version a day or two after this lands is the plan!

**Comment by lqd** [comment]
Maybe the process to help review could be made easier if we split adding the benchmark code into its own commit (and which could largely be ignored), and the manual changes for the json, readme, patches into different commits that are more valuable.

**Comment by bjorn3** [comment]
The total time being faster is likely due to cranelift-codegen-meta containing and generating less code than before. In the old version most of the individual backends was generated by cranelift-codegen-meta. In the new version most of the generated code is checked in and the code to generate it is only compiled when using a certain feature flag.

File: collector/benchmarks/cranelift-codegen-0.82.1/perf-config.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "category": "primary"
+}
```
Comment (nnethercote) [inline collector/benchmarks/cranelift-codegen-0.82.1/perf-config.json:3]: The `perf-config.json` for the old `cranelift-codegen` has a `cargo_toml` and a `touch_file` entry. Are they needed here too?

Comment (lqd) [reply collector/benchmarks/cranelift-codegen-0.82.1/perf-config.json:3]: I didn't expect them to be: the old benchmark is a workspace with in-tree members, i.e. likely a git clone of the entire cranelift repo, and I believed it used these settings to only profile `cranelift-codegen` amongst all the sub-crates. The new benchmark is just the crate itself from crates.io, and is not a workspace: the default behavior of profiling a library should work fine ?

Comment (nnethercote) [reply collector/benchmarks/cranelift-codegen-0.82.1/perf-config.json:3]: Makes sense, thanks!

**Review by nnethercote — APPROVED** [review]

File: collector/benchmarks/cranelift-codegen-0.82.1/perf-config.json (hunk 1 above)
Comment (lqd) [reply collector/benchmarks/cranelift-codegen-0.82.1/perf-config.json:3]: I will launch other runs of the benchmark, with a more realistic workload, to make sure

**Pull request merged**